File: packages/ts/src/output-modes.ts
// Протокольный ВОКАБУЛЯР формата текста (media-типы) — agent-facing SSOT.
//
// Этим словарём агент декларирует `defaultOutputModes` в agent-card. Сама content-negotiation
// (выбор формата/каталога) и enforcement — в хосте (`@ai37/agent-host`, src/output-modes.ts),
// т.к. это его зона ответственности; сюда вынесены только разделяемые константы media-типов.
//
// См. docs/ecosystem/v2/10-agui-protocol.md (РЕШЕНИЕ 10), client_capabilities.json (@a2ui/web_core).

/** Простой текст. */
export const OUTPUT_MODE_TEXT = 'text/plain'
/** Markdown. */
export const OUTPUT_MODE_MARKDOWN = 'text/markdown'
/** Markdown под рендерер SP-AI. */
export const OUTPUT_MODE_MARKDOWN_SPAI = 'text/vnd.markdown+spai-renderer'

/** Текстовые modes по убыванию «богатства». */
export const TEXT_OUTPUT_MODES = [
  OUTPUT_MODE_MARKDOWN_SPAI,
  OUTPUT_MODE_MARKDOWN,
  OUTPUT_MODE_TEXT,
] as const

const TEXT_MODE_SET = new Set<string>(TEXT_OUTPUT_MODES)
/** Является ли mode текстовым media-типом. */
export const isTextOutputMode = (mode: string): boolean => TEXT_MODE_SET.has(mode)

File: packages/python/src/ai37_agent_sdk/output_modes.py
"""Протокольный ВОКАБУЛЯР формата текста (media-типы) — agent-facing SSOT.

Зеркало TS-модуля ``@ai37/agent-sdk`` ``output-modes.ts``. Этим словарём агент
декларирует ``defaultOutputModes`` в agent-card. Сама content-negotiation (выбор
формата/каталога) и enforcement — в хосте (``ai37-agent-host``); сюда вынесены только
разделяемые константы media-типов.

См. docs/ecosystem/v2/10-agui-protocol.md (РЕШЕНИЕ 10).
"""

from __future__ import annotations

#: Простой текст.
OUTPUT_MODE_TEXT = "text/plain"
#: Markdown.
OUTPUT_MODE_MARKDOWN = "text/markdown"
#: Markdown под рендерер SP-AI.
OUTPUT_MODE_MARKDOWN_SPAI = "text/vnd.markdown+spai-renderer"

#: Текстовые modes по убыванию «богатства».
TEXT_OUTPUT_MODES = (
    OUTPUT_MODE_MARKDOWN_SPAI,
    OUTPUT_MODE_MARKDOWN,
    OUTPUT_MODE_TEXT,
)

_TEXT_MODE_SET = frozenset(TEXT_OUTPUT_MODES)


def is_text_output_mode(mode: str) -> bool:
    """Является ли ``mode`` текстовым media-типом."""
    return mode in _TEXT_MODE_SET
